src/test/kotlin/com/codingchallenge/transactionanalyser/TransactionLoaderTest.kt

```diff
@@ -0,0 +1,52 @@
+package com.codingchallenge.transactionanalyser
+
+import com.codingchallenge.transactionanalyser.domain.TransactionType
+import org.junit.jupiter.api.Assertions.*
+import org.junit.jupiter.api.Test
+import java.math.BigDecimal
+import java.time.LocalDateTime
+import java.time.format.DateTimeFormatter
+
+internal class TransactionLoaderTest{
+    private val transactionLoader = TransactionLoader(
+        this::class.java.classLoader.getResourceAsStream("transactions.csv") ?: throw IllegalArgumentException("File not found")
+    )
+    private val formatter = DateTimeFormatter.ofPattern("dd/MM/yyyy HH:mm:ss")
+
+    @Test
+    fun givenCsv_WhenLoadTransactionsFromCsv_ThenTheExpectedTransactionsReturned(){
```

Comment:
You will also see in many of our real prod code tests, we are big fans of using `@Nested` tests. so you could have one layer of given / when / then in a class hierarchy where a child class can run in a nested way but all tests still run.
Good example here worth reading: https://kotlintesting.com/nested-tests-in-junit5/

Reply:
That's handy.